components-rs/log.rs

```diff
@@ -211,3 +213,28 @@ pub extern "C" fn ddog_get_logs_count(level: CharSlice) -> u32 {
         *counter.entry(level).or_default()
     });
 }
+
+#[no_mangle]
+pub extern "C" fn ddog_get_integration_error_log() -> *const c_char {
+    return INTEGRATION_ERROR_LOGS.with(|logs| {
+        let mut logs = logs.borrow_mut();
+        if !logs.is_empty() {
+            match CString::new(logs.pop().unwrap().as_str()) {
+                Ok(c_string) => {
+                    c_string.into_raw()
+                }
+                Err(_) => std::ptr::null(),
+            }
+        } else {
+            std::ptr::null()
+        }
+    });
+}
+
+#[no_mangle]
+pub extern "C" fn ddog_add_integration_error_log(log: CharSlice) {
+    INTEGRATION_ERROR_LOGS.with(|logs| {
+        logs.borrow_mut().push(log.to_utf8_lossy().to_string());
+    });
+}
```

Comment:
Why do we need this at all, and cannot simply call `ddog_sidecar_telemetry_add_integration_log_buffer` everywhere? Why do we buffer this in a Vec here? (you'd move the logic from components/telemetry/telemetry.c to ext/telemetry.c, and call `ddog_sidecar_telemetry_add_integration_log_buffer(ddtrace_telemetry_buffer(), log_slice)` directly)

Reply:
My understanding is that we want to send the telemetry at the end of a request. Therefore the Vec vector is used to store the logs temporary to send it at the end. Adding directly to the log buffer may send the logs along the request.

If you think that is not the problem, your suggestion sounds indeed appropriate.

Reply:
I think that's not the problem. It adds complexity in the code for no benefit - what we want is no unnecessary submissions, i.e. not start the submission machinery for every single thing, it has some fixed overhead. But if we flush anyways, that's fine too.